nautobot/extras/querysets.py

```diff
@@ -29,19 +29,19 @@ def get_for_object(self, obj):
         device_redundancy_group = getattr(obj, "device_redundancy_group", None)
 
         # Get the group of the assigned tenant, if any
-        if obj._meta.model_name == "device":
-            tenant_group = obj.tenant.tenant_group if obj.tenant else None
-            tenant = obj.tenant if obj.tenant else None
-        else:
-            tenant_group = obj.cluster.tenant.tenant_group if obj.cluster.tenant else None
-            tenant = obj.cluster.tenant if obj.cluster.tenant else None
+        tenant_group = obj.tenant.tenant_group if obj.tenant else None
+        tenant = obj.tenant if obj.tenant else None
 
         if tenant_group:
             tenant_groups = tenant_group.ancestors(include_self=True)
         else:
             tenant_groups = []
         # Match against the directly assigned location as well as any parent locations
-        location = getattr(obj, "location", None)
+        if cluster is not None:
+            location = getattr(cluster, "location", None)
+        else:
+            location = getattr(obj, "location", None)
```

Comment:
This is actually an enhancement/behavior change, right? It looks like in 1.6 we weren't considering cluster location/site/region as a factor. Probably worth documenting in a change fragment.